internal/execution/executor.go

```diff
@@ -217,7 +219,19 @@ func (e *Executor) updateComposition(ctx context.Context, env *Env, oldComp *api
 	})
 }
 
-func skipSynthesis(comp *apiv1.Composition, env *Env) bool {
+func skipSynthesis(comp *apiv1.Composition, env *Env) (string, bool) {
 	synthesis := comp.Status.CurrentSynthesis
-	return synthesis == nil || synthesis.Synthesized != nil || synthesis.UUID != env.SynthesisUUID || (synthesis.Attempts > 0 && synthesis.Attempts > env.SynthesisAttempt)
+	if synthesis == nil {
```

Comment:
Tested a theory by adding better logging. Theory didn't end up being true, but thought the logging might be useful for the future